src/main/java/org/opensearch/knn/plugin/rest/RestSearchModelHandler.java

```diff
@@ -69,9 +70,22 @@ private void checkUnSupportedParamsExists(RestRequest request) {
         throw new IllegalArgumentException(errorMessage);
     }
 
+    private void searchSizeValidation(RestRequest request) {
+        ValidationException validationException = new ValidationException();
+        if (request.hasParam("size")) {
+            int searchSize = request.paramAsInt("size", 1);
+            if (searchSize > 1000 || searchSize <= 0) {
+                String errorMessage = "Size must be > 0 and <= 1000";
```

Comment:
Error message text looks a bit raw, do you know if we've used similar symbols in other messages. If not I suggest we change it to something like "size must be between 0 and 100 inclusive"